tests/modules/lib/cddl-gen/CMakeLists.txt

```diff
@@ -12,7 +12,21 @@ project(test_cddl_gen)
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
 
-target_cddl_source(app test.cddl ENTRY_TYPES Test ENCODE
-  ${VERBOSE}
-  ${CANONICAL}
+set(py_command
+  cddl-gen
+  -c ${CMAKE_CURRENT_LIST_DIR}/test.cddl
+  code
+  --output-cmake ${PROJECT_BINARY_DIR}/test.cmake
+  -t Test
+  -e
   )
+
+execute_process(
+  COMMAND ${py_command}
+  COMMAND_ERROR_IS_FATAL ANY
+  )
+
+include(${PROJECT_BINARY_DIR}/test.cmake)
```

Comment:
so you are generating a cmake file, but there is no comment of what this file does.

Also, why is a CMake file created ?

From build system perspective, I really don't like the priciple of generating CMake files, especially not defining new library targets.
The libraries created by the build system should be directly visible when examine the build system.
I have no possible way I can review the build code generated this way when readin a PR.

Side note: why is this feature not described in the help from cddl-gen ?
```
$ cddl-gen --version
cddl-gen 0.3.0
$ cddl-gen --help
usage: cddl-gen [-h] [--version] -c CDDL [--default-max-qty DEFAULT_MAX_QTY] [-v] {code,convert} ...

Parse a CDDL file and validate/convert between YAML, JSON, and CBOR. Can also generate C code for validation/encoding/decoding of CBOR.

positional arguments:
  {code,convert}

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  -c CDDL, --cddl CDDL  Path to one or more input CDDL file(s). Passing multiple files is equivalent to concatenating them.
  --default-max-qty DEFAULT_MAX_QTY, --dq DEFAULT_MAX_QTY
                        Default maximum number of repetitions when no maximum is specified. This is needed to construct complete C types. This is relevant for the generated code. It is not relevant for converting, except when handling
                        data that will be decoded by generated code. The default value of this option is 3. Set it to a large number when not relevant.
  -v, --verbose         Print more information while parsing CDDL and generating code.
```

Reply:
You're right, this isn't mentioned in the help text, only in the README. I made a task (NCDSK-12904) to update the text of `cddl-gen code --help`. Thanks.

Meanwhile, I changed the tests to not use the generated cmake files since it's partially unneeded when we have `nrf/modules/cddl_gen/CMakeLists.txt`

Reply:
> Meanwhile, I changed the tests to not use the generated cmake files since it's partially unneeded when we have `nrf/modules/cddl_gen/CMakeLists.txt`

Does this mean you still consider to propose using generated CMake files in NCS later ?
I'm very reluctant to approve that, for several reasons, like given here:
https://github.com/nrfconnect/sdk-nrf/pull/6344#discussion_r766596196
https://github.com/nrfconnect/sdk-nrf/pull/6344#pullrequestreview-828755837

but also because the `execute_process()` function doesn't provide a `OUTPUT` and `DEPENDS` arguments.

With using `add_custom_command()` having OUTPUT, then a `ninja/make clean`  knows how to correctly remove generated sources.
And the DEPENDS ensures that sources will be correctly regenerated when `test.cddl` is updated.

Reply:
>Does this mean you still consider to propose using generated CMake files in NCS later ?

I have no such plans.

Changed to using `add_custom_command()`, thanks.